roles/tests-integration/tests/pool_integration.rs

```diff
@@ -0,0 +1,45 @@
+mod common;
+
+use const_sv2::{
+    MESSAGE_TYPE_COINBASE_OUTPUT_DATA_SIZE, MESSAGE_TYPE_NEW_TEMPLATE,
+    MESSAGE_TYPE_SETUP_CONNECTION, MESSAGE_TYPE_SETUP_CONNECTION_SUCCESS,
+    MESSAGE_TYPE_SET_NEW_PREV_HASH,
+};
+use roles_logic_sv2::{
+    common_messages_sv2::{Protocol, SetupConnection},
+    parsers::{CommonMessages, PoolMessages},
+};
+
+#[tokio::test]
```

Comment:
Shouldn't we add a description of the test? I think it's a good practice since we are going to have a lot of them